liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -42,38 +45,84 @@
     
     private ServerXmlDocument() {    
     }
+
+    /**
+     * @param filePath
+     * @param fileName application file name
+     * @return true if the application with the file name is configured.
+     * @throws Exception
+     */
+    private static boolean isFoundWebAppWithFileName(String filePath, String fileName) throws Exception {
+        return isFoundAppConfigByTagAndFileName(filePath, "webApplication", fileName);
+    }
     
-    public static boolean isFoundTagName(String fileName, String tag) throws Exception {
+    /**
+     * @param filePath 
+     * @param tag specify element tag to parse
+     * @param fileName application file name
+     * @return true if the application with the file name within the specified element tag is configured. 
+     * @throws Exception
+     */
+    private static boolean isFoundAppConfigByTagAndFileName(String filePath, String tag, String fileName) throws Exception {
 
-        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-        DocumentBuilder builder = factory.newDocumentBuilder();
-        Document doc = builder.parse(fileName);
+        NodeList appList = getNodeList(filePath, "/server/" + tag);
+        
+        if (appList == null || appList.getLength() == 0) {
+            return false; 
+        }
+        else {
+          if (fileName == null) {
+            return true;
+          }
+        } 
         
-        NodeList appList = doc.getElementsByTagName(tag);
+        for (int i = 0; i < appList.getLength(); i++) {
+            Node locNode = appList.item(i).getAttributes().getNamedItem("location");
+            
+            if (locNode != null) {
+                String locValue = locNode.getNodeValue();
+                if (locValue.equals(fileName)) {
+                    return true;
+                }
+            }
+        }
         
-        return appList != null && appList.getLength() > 0;
+        return false; 
     }
     
-    public static boolean isFoundWebApplication(String filePath, File configDirectory) throws Exception {
+    public static boolean isFoundAppConfig(String filePath, File configDirectory, String fileName) throws Exception {
```

Comment:
This method is too complicated.  Can you refractor it into a few methods and use 
``` code
if (condition || condition || ...)  {
    return true;
}
```
instead of 
```
... some logic...
if (condition) {
   return true;
}
... some logic ...
if (condition) {
   return true;
}
... some logic ...
if (condition) {
   return true;
}
```